### PR TITLE
[Reviewer: Ellie] Adds support for enterprise alarms MIB

### PR DESCRIPTION
**Pull request by AME2**
Changes here include updating the Python JSON parser to test that the new enterprise MIB fields are under 4096 characters.

### CODE REVIEW

File: metaswitch/common/alarms_parser.py [hunk 1]
```diff
@@ -92,6 +92,31 @@ def parse_alarms_file(json_file):
...
+                # The extended details and extended descriptions fields are
+                # optional. We should only check they are under 4096 characters
+                # in the case where they exist.
+                try:
+                    level['extended details']
+                except KeyError:
+                    # Valid to not have extended details
+                    pass
+                else:
+                    assert len(level['extended details']) < 4096, \
```
Comment (eleanor-merry) [inline metaswitch/common/alarms_parser.py]: You can just put this test in the try

File: metaswitch/common/alarms_parser.py [hunk 2]
```diff
@@ -92,6 +92,31 @@ def parse_alarms_file(json_file):
...
+        "Extended details length was greater than 4096 characters in alarm {}".format(name)
+                try:
+                    level['extended description']
+                except KeyError:
+                    # Valid to not have an extended description
+                    pass
+                else:
+                    assert len(level['extended description']) < 4096, \
+        "Extended description length was greater than 4096 characters in alarm {}".format(name)
```
Comment (eleanor-merry) [inline metaswitch/common/alarms_parser.py]: You've got some unnecessary checks now for the cause/effect/action below

File: metaswitch/common/test/alarms_parser.py [hunk 3]
```diff
@@ -43,6 +43,14 @@ def testValidFile(self):
...
         self.assertIn(CRITICAL, test_alarm[2], msg="No critical state.")
 
+    def testExtendedValidFile(self):
+        alarms = parse_alarms_file('metaswitch/common/test/test_extended_alarm.json')
+        test_alarm = alarms[0]
+        self.assertEqual(test_alarm[0], 'NAME', msg="Incorrect name.")
+        self.assertEqual(test_alarm[1], 1000, msg="Incorrect index.")
+        self.assertIn(CLEARED, test_alarm[2], msg="No cleared state.")
+        self.assertIn(CRITICAL, test_alarm[2], msg="No critical state.")
+
```
Comment (eleanor-merry) [inline metaswitch/common/test/alarms_parser.py]: No new UTs for too long parameters?

**Comment by eleanor-merry** [comment]
Looks good - I'm put some picky comments inline and I think you can beef up the UTs

File: metaswitch/common/test/test_extended_alarm.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (eleanor-merry) [inline metaswitch/common/test/test_extended_alarm.json]: This file isn't in any tests

File: metaswitch/common/alarms_parser.py [hunk 5]
```diff
@@ -92,12 +92,28 @@ def parse_alarms_file(json_file):
...
-                level['cause']
-                level['effect']
-                level['action']
+                assert len(level['cause']) < 4096, \
+     "Cause length was greater than 4096 characters in alarm {}".format(name)
+                assert len(level['effect']) < 4096, \
+     "Effect length was greater than 4096 characters in alarm {}".format(name)
+                assert len(level['action']) < 4096, \
+     "Action length was greater than 4096 characters in alarm {}".format(name)
+                # The extended details and extended descriptions fields are
```
Comment (eleanor-merry) [inline metaswitch/common/alarms_parser.py]: Please put a blank line above this comment

**Comment by eleanor-merry** [comment]
Generally good, two comments inline

File: metaswitch/common/test/test_extended_alarm.json (hunk 4 above)
Comment (AME2) [inline metaswitch/common/test/test_extended_alarm.json]: I can't remember what this means.
The test that uses the above JSON file is testExtendedValidFile, this checks the the JSON parser has correctly parsed the name, index and two severities of the alarm. I don't think there are any wrong JSON names.

Comment (eleanor-merry) [inline metaswitch/common/test/test_extended_alarm.json]: extended details and descriptions don't use the underscore.

**Pull request merged**